spec/rubocop/cop/lint/unneeded_require_statement_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Lint::UnneededRequireStatement do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  it "registers an offense when using `require 'enumerator'`" do
+    expect_offense(<<-RUBY.strip_indent)
+      require 'enumerator'
+      ^^^^^^^^^^^^^^^^^^^^ Remove unnecessary require statement.
+    RUBY
+  end
+
+  it 'autocorrects remove unnecessary require statement' do
```

Comment:
Might be good with a test case that checks that it doesn't leave an empty line where the `require` used to be. 🙂

Reply:
Commit c275b39 has improved test case and implementation.